joi-validations/users.validations.js

```diff
@@ -12,6 +13,21 @@ const UserValidation = {
       limit: number,
       offset: number
     }
+  },
+  checkAllowDm: {
+    query: {
+      source: string.required(),
+      post_id: uuid.when('source', {
+        is: 'post',
+        then: Joi.required(),
+        otherwise: Joi.optional()
+      }),
+      comment_id: uuid.when('source', {
+        is: 'comment',
+        then: Joi.required(),
+        otherwise: Joi.optional()
+      })
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new validation schema for `checkAllowDm` looks good. It correctly uses conditional validation based on the `source` field to require either `post_id` or `comment_id`. However, please make sure that the `source` field only accepts 'post' or 'comment' as valid values to prevent potential misuse.
```diff
- source: string.required(),
+ source: Joi.string().valid('post', 'comment').required(),
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->